eng/common/TestResources/Remove-TestResources.ps1

```diff
@@ -137,9 +137,31 @@ if (![string]::IsNullOrWhiteSpace($ServiceDirectory)) {
     }
 }
 
+$verifyDeleteScript = {
+    try {
+        $group = Get-AzResourceGroup -name $ResourceGroupName
+    } catch {
+        if ($Error[0].ToString().Contains("Provided resource group does not exist")) {
```

Comment:
Tip: the `$Error` stream isn't great if anything is running concurrent. Instead, catch an exception and check it's message to be explicit:

```powershell
  // ...
} catch {
  if ($_.ToString().Contains(...)) // ...
}
```

`$_` will be an `ErrorRecord`.

Reply:
Ah there we go. My powershell is a bit rusty. I had been trying `$_` in my local shell but not sure why it didn't work.

Reply:
It works for both thrown exceptions and cmdlet-written errors, both of which results in an `ErrorRecord` regardless:
```powershell
try { resolve-path foo -ea stop } catch { "Caught exception: " + $_.ToString() }
try { throw "exception message goes here" } catch { "Caught exception: " + $_.ToString() }
```